From: Naushir Patuck <naush@raspberrypi.com>
Date: Thu, 30 Jan 2020 12:35:44 +0000
Subject: staging: vc04_services: Fix vcsm overflow bug when counting
 transactions

The response block and local state were using u16 and u32 respectively
to represent transaction id.  When the former would wrap, there is a
mismatch and subsequent transactions will be marked as failures.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
---
 drivers/staging/vc04_services/vc-sm-cma/vc_sm_cma_vchi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/vc04_services/vc-sm-cma/vc_sm_cma_vchi.c b/drivers/staging/vc04_services/vc-sm-cma/vc_sm_cma_vchi.c
index 1455186a5e22..dea53c3d1618 100644
--- a/drivers/staging/vc04_services/vc-sm-cma/vc_sm_cma_vchi.c
+++ b/drivers/staging/vc04_services/vc-sm-cma/vc_sm_cma_vchi.c
@@ -34,7 +34,7 @@ struct sm_cmd_rsp_blk {
 	/* To be signaled when the response is there */
 	struct completion cmplt;
 
-	u16 id;
+	u32 id;
 	u16 length;
 
 	u8 msg[VC_SM_MAX_MSG_LEN];
